layer/clients/project_service.py

```diff
@@ -81,6 +84,29 @@ def get_project_by_id(self, project_id: UUID) -> Optional[Project]:
             raise generate_client_error_from_grpc_error(err, "internal")
         return None
 
+    def delete_account_projects_older_than(
+        self, account_id: uuid.UUID, date: datetime, dry_run=True
+    ) -> int:
+        try:
+            resp: GetAccountProjectsByIdResponse = self._service.GetAccountProjectsById(
+                GetAccountProjectsByIdRequest(
+                    account_id=AccountId(value=str(account_id))
+                )
+            )
+
+            def is_old_project(p: ProjectMessage) -> bool:
+                return p.created_time.ToDatetime() < date
+
+            deleted = 0
+            for p in list(resp.project):
+                if is_old_project(p):
+                    if not dry_run:
+                        self.remove_project(UUID(p.id.value))
```

Comment:
Do we want to catch the exception here, log and continue?

Otherwise there is a race between multiple runs getting the same list and then deleting orgs.

Reply:
Also can we filter test projects only?

I know some folks like @volkangurel like running tests with their own account (and I guess nothing stops an external person of doing so). We probably don't want to nuke all their projects in that case.

Reply:
yes good point, gonna filter on names like we do for accounts